src/main/java/ch/njol/skript/expressions/ExprVectorFromDirection.java

```diff
@@ -0,0 +1,102 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Direction;
+import ch.njol.util.Kleenean;
+import org.bukkit.Location;
+import org.bukkit.event.Event;
+import org.bukkit.util.Vector;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Name("Vectors - Create from Direction")
+@Description({
+	"Creates vectors from given directions.",
+	"Relative directions are relative to the origin, (0, 0, 0). Therefore, the vector from the direction 'forwards' is (0, 0, 1)."
+})
+@Examples({
+	"set {_v} to vector from direction upwards",
+	"set {_v} to vector in direction of player",
+	"set {_v} to vector in horizontal direction of player",
+	"set {_v} to vector from facing of player",
+	"set {_v::*} to vectors from north, south, east, and west"
+})
+@Since("INSERT VERSION")
+public class ExprVectorFromDirection extends SimpleExpression<Vector> {
+
+	static {
+		Skript.registerExpression(ExprVectorFromDirection.class, Vector.class, ExpressionType.SIMPLE,
+				"vector[s] [from|in] [direction] %directions%",
```

Comment:
This is intentional, to allow things like  ``vector from {_direction}`` instead of ``vector {_direction}`` (quite bad).

Reply:
`vector in in`, `vector in direction in the direction of player from target block` looks bad

Reply:
Sure, that's why it's optional.

Reply:
There's no good way to tell what kind of expression follows it, as far as I know, so providing more tools is better than providing too few IMO.

Reply:
`` in the direction of player from target block vector`` is also ugly, but it exists to allow `north vector`.
It's really a problem with directions in general being really ugly.

Reply:
See https://discord.com/channels/135877399391764480/836220422223036467/1083534586681753730 for me and pikachu discussing this.

Reply:
Pikachu was literally on the same page as me. At this point though the direction expression is behind the word `vector` which is a proper placement rather than it being in front, as all of Skript direction accepting patterns have it after.

Reply:
I'm willing to change it to `vector [from [direction]] %directions%` or even `vector [from] %directions%` but I'm not budging on at least keeping `from`. If someone's determined to make their code as ugly as possible, they can use it improperly. However, I'm unconvinced that there's any real benefit to removing it, only loss of functionality.

Reply:
The only ugly outcome is using a variable. This is how Skript has been when you provide any direction expression. Look at other direction expression using expressions. If you're using a variable for a direction at that point, you're already trying to be tricky.

Reply:
Yeah, it's specifically for variables and and things like the ExprFacing. I understand that Skript often has directions directly after syntaxes (usually in combination with a location, which isn't happening here), but in this case I don't think that's appropriate (honestly it could be improved in those places, too, in my opinion). I see no real benefit to removing `from`. It's not required, so it's not like ``vector from in direction of entity`` is somehow required over ``vector in direction of entity``. There are plenty of ugly syntaxes without it, like ``vector left of``, so it's not like it's making anything drastically worse. And yet it makes some combinations much nicer. To me, it seems like a benefit with no real downsides, and I still fail to see why removing it would be any better.

Plus, using a variable that's a direction is hardly being tricky. I've done it numerous times in scripts ranging from conveyor belts to fill algorithms.

Reply:
I would go with Sovdeeth's current implementation, as he said having more is better than less. In addition to that, some places in skript do have this kind weird syntaxes (see example below), it's fine as long as it's not compulsory and if used the way you mentioned it would be the user misusing the syntax.

Taking for example `ExprElementOf` (`[the] %number%(st|nd|rd|th)`)
![image](https://user-images.githubusercontent.com/20037329/233348359-df84c9ef-f44d-4d51-8a19-317b2bad8f10.png)